### PR TITLE
feat(v4): Compose a new conversation from a phone number.

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

This PR adds the ability to create new contacts from phone numbers in the compose conversation form.

Fixes https://linear.app/chatwoot/issue/CW-3781/ability-to-create-new-conversation-contacts-from-phone-numbers-in

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

**Loom video**
https://www.loom.com/share/d9e16714c66d4095acccedac1f75e876?sid=10f4db7b-886f-4e01-a9c6-82513a8438e2

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by sojan-official — APPROVED** [review]
Tested the functionality on staging and working fine !!! 🔥

File: app/javascript/dashboard/components-next/NewConversation/helpers/composeConversationHelper.js [hunk 1]
```diff
@@ -80,6 +80,11 @@ export const getCapitalizedNameFromEmail = email => {
   return name.charAt(0).toUpperCase() + name.slice(1);
 };
 
+export const getNameFromPhoneNumber = phoneNumber => {
+  // Remove the '+' prefix if it exists
+  return phoneNumber.startsWith('+') ? phoneNumber.slice(1) : phoneNumber;
```
Comment (scmmishra) [inline app/javascript/dashboard/components-next/NewConversation/helpers/composeConversationHelper.js:85]: This check is already done on line 203 in createNewContact, so it's kinda redundant. Minor nit-pick

File: app/javascript/dashboard/components-next/NewConversation/helpers/composeConversationHelper.js [hunk 2]
```diff
@@ -193,10 +198,12 @@ export const searchContacts = async ({ keys, query }) => {
   return filteredPayload || [];
 };
 
-export const createNewContact = async email => {
+export const createNewContact = async input => {
   const payload = {
-    name: getCapitalizedNameFromEmail(email),
-    email,
+    name: input.startsWith('+')
```
Comment (scmmishra) [inline app/javascript/dashboard/components-next/NewConversation/helpers/composeConversationHelper.js:203]: Better to check using regex, something like this `[+0-9\s()]+`

**Review by scmmishra — CHANGES_REQUESTED** [review]
Tiny suggestions

**Review by scmmishra — APPROVED** [review]

File: app/javascript/dashboard/components-next/NewConversation/helpers/composeConversationHelper.js (hunk 1 above)
Comment (iamsivin) [reply app/javascript/dashboard/components-next/NewConversation/helpers/composeConversationHelper.js]: Done https://github.com/chatwoot/chatwoot/pull/10568/commits/038fd433ccefa2199b9e9cc0b314309904eaf1cc

File: app/javascript/dashboard/components-next/NewConversation/helpers/composeConversationHelper.js (hunk 2 above)
Comment (iamsivin) [reply app/javascript/dashboard/components-next/NewConversation/helpers/composeConversationHelper.js:198]: We had a discussion on this, so I’m resolving this comment.

**Pull request merged**